### PR TITLE
Prac 05 code review request

### PR DESCRIPTION
**Pull request by gaylejuntilla**
Sorry for the late request @Dav0-12 , I would appreciate it if you could check out my code, specifically structure wise(if I chose the right method/function) or any variable naming since I find it a bit tricky still. Thank you!

### CODE REVIEW

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""Email to name program"""
+
+
+def main():
+    """Create and display dictionary of emails and names"""
+    email_to_name = {}
```
Comment (Dav0-12) [inline prac_05/emails.py:6]: Good name

File: prac_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""Email to name program"""
+
```
Comment (Dav0-12) [inline prac_05/emails.py:2]: Looks good, I don't have any problems with this : )

File: prac_05/hex_colours.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+
+COLOUR_TO_CODE = {"absolutezero": "#0048ba", "acidgreen": "#b0bf1a", "aliceblue": "	#f0f8ff", "amaranth": "	#e52b50", "amber": "#ffbf00",
+                  "amethyst": "#9966cc", "antiquewhite": "#faebd7", "apricot": "#fbceb1", "armygreen": "#4b5320", "asparagus": "#87a96b"}
+
+colour_name = input("Colour name: ").lower()
```
Comment (Dav0-12) [inline prac_05/hex_colours.py:6]: good use of .lower()

File: prac_05/state_names.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    except KeyError:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+for state, state_name in CODE_TO_NAME.items():
+    max_state_length = max(len(state) for state in CODE_TO_NAME.keys())
+    max_sate_name_length = max(len(state_name) for state_name in CODE_TO_NAME.values())
+    print(f"{state:{max_state_length}} is {state_name:{max_sate_name_length}}")
+
+
```
Comment (Dav0-12) [inline prac_05/state_names.py]: Your code is really good, I'm just scraping the bottom of the barrel here. In PEP8 they like it when there is only one blank line at the end of a document but no issue : )

File: prac_05/state_names.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+state_code = input("Enter short state: ").upper()
+while state_code != "":
+    try:
+        print(state_code, "is", CODE_TO_NAME[state_code])
+    except KeyError:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+for state, state_name in CODE_TO_NAME.items():
+    max_state_length = max(len(state) for state in CODE_TO_NAME.keys())
```
Comment (Dav0-12) [inline prac_05/state_names.py]: These list comprehension functions will go through all the items in CODE_TO_NAME and find the longest state and state name. Do they have to be in a for loop then...

File: prac_05/state_names.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    max_sate_name_length = max(len(state_name) for state_name in CODE_TO_NAME.values())
```
Comment (Dav0-12) [inline prac_05/state_names.py]: Just a small typo here with sate, nothing major

File: prac_05/wimbledon.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
+CHAMPION_INDEX = 2
+COUNTRY_INDEX = 1
+
+
+def main():
+    """Process wimbledon.csv and display champions and countries"""
```
Comment (Dav0-12) [inline prac_05/wimbledon.py:6]: Really clear docstring 👍

File: prac_05/wimbledon.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
+CHAMPION_INDEX = 2
```
Comment (Dav0-12) [inline prac_05/wimbledon.py:1]: I like that you define the index here 10/10. It removes the weird magic numbers in the code

File: prac_05/wimbledon.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    data = []
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split(",")
+            data.append(parts)
+        return data
+
+
+main()
```
Comment (Dav0-12) [inline prac_05/wimbledon.py:46]: Don't forget your blank line at the end of the document 😸

File: prac_05/word_occurrrences.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    try:
+        word_to_count[word] += 1
+    except KeyError:
+        word_to_count[word] = 1
+
+ordered_words = sorted(word_to_count.keys())
+max_word_length = max(len(word) for word in words)
+for word in ordered_words:
+    print(f"{word:{max_word_length}} : {word_to_count[word]}")
+
```
Comment (Dav0-12) [inline prac_05/word_occurrrences.py]: Really well done 👍

**Review by Dav0-12 — COMMENTED** [review]
Hey @gaylejuntilla, no problems with the late review I get it. Your code is looking really good, only a few very pedantic things to note but your structure is great. You should be proud of it. 👍 🌮

**Comment by gaylejuntilla** [comment]
> Hey @gaylejuntilla, no problems with the late review I get it. Your code is looking really good, only a few very pedantic things to note but your structure is great. You should be proud of it. 👍 🌮

Thanks for your review @Dav0-12 it was much appreciated :)) especially that for loop check, completely went over my head ahaha

**Pull request merged**